### PR TITLE
Elaborated on various areas of authentication documentation

### PR DESCRIPTION
**Pull request by AE-MS**
## Description

Updated various parts of the `authentication` capability documentation to make it clearer (hopefully!)

### Validation performed:

I generated the docs locally, proofread them, and validated that all the links appear to work.

### CODE REVIEW

**Review by jekloudaMSFT — DISMISSED** [review]
Thanks for updating these!

File: packages/teams-js/src/public/authentication.ts [hunk 1]
```diff
@@ -28,39 +33,53 @@ export namespace authentication {
...
   /**
    * @deprecated
    * As of 2.0.0, please use {@link authentication.authenticate authentication.authenticate(authenticateParameters: AuthenticatePopUpParameters): Promise\<string\>} instead.
    *
-   * Initiates an authentication request, which opens a new window with the specified settings.
-   *
-   * @remarks
-   * The authentication flow must start and end from the same domain, otherwise success and failure messages won't be returned to the window that initiated the call.
+   * The documentation for {@link authentication.authenticate authentication.authenticate(authenticateParameters: AuthenticatePopUpParameters): Promise\<string\>} applies
+   * to this function. Please see that documentation.
```
Comment (erikadoyle) [inline packages/teams-js/src/public/authentication.ts:78]: Can probably omit "Please see that documentation." sentence, as its already implied by the overall comment

File: packages/teams-js/src/public/authentication.ts [hunk 2]
```diff
@@ -153,20 +172,26 @@ export namespace authentication {
   }
 
   /**
-   * Requests an Azure AD token to be issued on behalf of the app. The token is acquired from the cache
-   * if it is not expired. Otherwise a request is sent to Azure AD to obtain a new token.
+   * Requests an Azure AD token to be issued on behalf of your app in an SSO flow.
+   * The token is acquired from the cache if it is not expired. Otherwise a request is sent to Azure AD to
+   * obtain a new token.
+   * This function is used to enable SSO scenarios. Please see [this page](https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/authentication/tab-sso-overview)
+   * for more detail.
```
Comment (erikadoyle) [inline packages/teams-js/src/public/authentication.ts:179]: How about:

See [Eanble SSO for tab apps](https://learn.microsoft.com/microsoftteams/platform/tabs/how-to/authentication/tab-sso-overview) for more details.

File: packages/teams-js/src/public/authentication.ts [hunk 3]
```diff
@@ -28,39 +33,53 @@ export namespace authentication {
...
-   * Initiates an authentication request, which opens a new window with the specified settings.
+   * Initiates an authentication flow which requires a new window.
+   * There are two primary uses for this function:
+   * 1. When your app needs to authenticate using a 3rd-party identity provider (not Azure Active Directory)
+   * 2. When your app needs to show authentication UI that is blocked from being shown in an iframe (e.g., Azure Active Directory consent prompts)
+   *
+   * You can learn more details about this flow [here](https://learn.microsoft.com/microsoftteams/platform/tabs/how-to/authentication/auth-flow-tab)
+   *
+   * This function is *not* needed for "standard" Azure SSO usage. Using {@link getAuthToken} is usually sufficient in that case (learn more
+   * [here](https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/authentication/tab-sso-overview))
```
Comment (erikadoyle) [inline packages/teams-js/src/public/authentication.ts:55]: How about:

`Using {@link getAuthToken} is usually sufficient in that case. For more, see [Eanble SSO for tab apps](https://learn.microsoft.com/microsoftteams/platform/tabs/how-to/authentication/tab-sso-overview).`

File: packages/teams-js/src/public/authentication.ts [hunk 4]
```diff
@@ -28,39 +33,53 @@ export namespace authentication {
...
   }
 
   /**
-   * Initiates an authentication request, which opens a new window with the specified settings.
+   * Initiates an authentication flow which requires a new window.
+   * There are two primary uses for this function:
+   * 1. When your app needs to authenticate using a 3rd-party identity provider (not Azure Active Directory)
+   * 2. When your app needs to show authentication UI that is blocked from being shown in an iframe (e.g., Azure Active Directory consent prompts)
+   *
+   * You can learn more details about this flow [here](https://learn.microsoft.com/microsoftteams/platform/tabs/how-to/authentication/auth-flow-tab)
```
Comment (erikadoyle) [inline packages/teams-js/src/public/authentication.ts:52]: How about: 

`For more details, see [Enable authentication using third-party OAuth provider](https://learn.microsoft.com/microsoftteams/platform/tabs/how-to/authentication/auth-flow-tab).`

File: packages/teams-js/src/public/authentication.ts [hunk 5]
```diff
@@ -28,39 +33,53 @@ export namespace authentication {
...
+   *
+   * This function is *not* needed for "standard" Azure SSO usage. Using {@link getAuthToken} is usually sufficient in that case (learn more
+   * [here](https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/authentication/tab-sso-overview))
    *
    * @remarks
    * The authentication flow must start and end from the same domain, otherwise success and failure messages won't be returned to the window that initiated the call.
+   * As described on the [3rd party OAuth provider documentation](https://learn.microsoft.com/microsoftteams/platform/tabs/how-to/authentication/auth-flow-tab), the
+   * authentication flow starts and ends at an endpoint on your own service (with a redirect round-trip to the 3rd party identity provider in the middle).
```
Comment (erikadoyle) [inline packages/teams-js/src/public/authentication.ts:60]: To tighten up the text, how about:

`The [Teams authentication flow](https://learn.microsoft.com/microsoftteams/platform/tabs/how-to/authentication/auth-flow-tab) starts and ends at an endpoint on your own service (with a redirect round-trip to the 3rd party identity provider in the middle).`

**Review by erikadoyle — COMMENTED** [review]
Thank you! These are great improvements to the docs. Added a few suggestions to tighten up the text.

I checked the new links from a local Typedoc build--everything resolved correctly.

File: packages/teams-js/src/public/authentication.ts [hunk 6]
```diff
@@ -153,20 +172,26 @@ export namespace authentication {
   }
 
   /**
-   * Requests an Azure AD token to be issued on behalf of the app. The token is acquired from the cache
-   * if it is not expired. Otherwise a request is sent to Azure AD to obtain a new token.
+   * Requests an Azure AD token to be issued on behalf of your app in an SSO flow.
+   * The token is acquired from the cache if it is not expired. Otherwise a request is sent to Azure AD to
+   * obtain a new token.
+   * This function is used to enable SSO scenarios. See [Enable SSO for tab apps](https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/authentication/tab-sso-overview)
```
Comment (erikadoyle) [inline packages/teams-js/src/public/authentication.ts:178]: Whoops, stray `en-us` in the path

**Review by erikadoyle — DISMISSED** [review]
Looks great!

**Review by jekloudaMSFT — APPROVED** [review]

**Review by erikadoyle — APPROVED** [review]

**Review by jekloudaMSFT — APPROVED** [review]

**Pull request merged**